praw/objector.py

```diff
@@ -61,6 +71,41 @@ def objectify(self, data):
                 or {'id', 'name', 'permissions'}.issubset(set(data.keys()))):
             parser = self.parsers[self._reddit.config.kinds['redditor']]
             return parser.parse(data, self._reddit)
+        elif isinstance(data, dict) and (
+                {'conversation', 'messages', 'user', 'modActions'}
+                .issubset(set(data.keys()))):
+            # Modmail conversation
+            parser = self.parsers['ModmailConversation']
+            return parser.parse(data, self._reddit)
+        elif isinstance(data, dict) and 'actionTypeId' in data.keys():
+            # Modmail mod action
+            data = self.fix_dict_keys(data, {'actionTypeId': 'action_type'})
+            parser = self.parsers['ModmailAction']
+            return parser.parse(data, self._reddit)
+        elif isinstance(data, dict) and 'isInternal' in data.keys():
+            # Modmail message
+            camel_attributes = {'bodyMarkdown': 'body_markdown',
+                                'isInternal': 'is_internal'}
+            data = self.fix_dict_keys(data, camel_attributes)
+            parser = self.parsers['ModmailMessage']
+            return parser.parse(data, self._reddit)
+        elif isinstance(data, dict) and 'isAdmin' in data.keys():
+            # Modmail author
+            camel_attributes = {'isAdmin': 'is_admin',
+                                'isDeleted': 'is_deleted',
+                                'isHidden': 'is_hidden',
+                                'isMod': 'is_mod',
+                                'isOp': 'is_op',
+                                'isParticipant': 'is_participant'}
+            data = self.fix_dict_keys(data, camel_attributes)
+            del(data['id'])
+            parser = self.parsers[self._reddit.config.kinds['redditor']]
+            return parser.parse(data, self._reddit)
+        elif isinstance(data, dict) and 'displayName' in data.keys():
+            # Modmail subreddit
+            data = self.fix_dict_keys(data, {'displayName': 'display_name'})
+            parser = self.parsers[self._reddit.config.kinds[data['type']]]
+            return parser.parse(data, self._reddit)
         elif isinstance(data, dict) and 'user' in data:
```

Comment:
With the camelCase conversion being handled automatically, most of the new cases just look like "If 'foo' is in the dict, use bar as the parser," which could easily be factored out. The catch is with keys that have to be manually modified — `del data['id']` is the one that's come up so far, but it looks like there are more when it comes to the `user` part of the response.